Applications/Spire/Include/Spire/Ui/RealSpinBox.hpp

```diff
@@ -0,0 +1,128 @@
+#ifndef SPIRE_REAL_SPIN_BOX_HPP
+#define SPIRE_REAL_SPIN_BOX_HPP
+#include <boost/multiprecision/cpp_dec_float.hpp>
+#include <boost/optional.hpp>
+#include <QAbstractSpinBox>
+#include <QLocale>
+#include <QRegularExpression>
+#include <QStyle>
+#include "Spire/Spire/Spire.hpp"
+
+namespace Spire {
+
+  //! Represents a widget for entering numeric values.
+  class RealSpinBox : public QAbstractSpinBox {
+    public:
+
+      //! The maximum number of accepted decimal places.
+      static const auto MAX_DECIMAL_PLACES = 15;
+
+      //! Numeric type used by the widget.
+      using Real = boost::multiprecision::cpp_dec_float<MAX_DECIMAL_PLACES>;
```

Comment:
The term "decimal place" is being used incorrectly here, change it to ```PRECISION```. The term "decimal places" refers to the position of a digit to the right of a decimal point, however, ```cpp_dec_float``` takes as a parameter the precision, not the decimal places. The precision refers to the number of digits that are represented. Precision and decimal places are not the same thing. For example a Nexus::Quantity and a Nexus::Money have 15 decimal places of precision with 6 decimal places, so it's possible to exactly represent numbers in the range [0.000001, 999999999.999999], which means a number with a leading "1", followed by fourteen "0"s, followed by a "1", which is  "1000000000000001" can not be represented even though it has zero decimal places. That number requires 16 digits of precision but Nexus::Quantity only guarantees 15 digits of precision.

The difference between precision, decimal places, and general matters involving floating point representation is worth having a basic understanding of, so feel free to take some time to learn about it. You don't have to be an expert in it but having a basic understanding will be valuable.